File: test51_out/pollen.net.transport/SingleBufferManager/SingleBufferManager.js

/*
 * ======== MODULE pollen.net.transport.SingleBufferManager ========
 */

var $$c = function() {
    const SingleBufferManager = {};
    SingleBufferManager.SingleBufferManager = SingleBufferManager
    SingleBufferManager.$name = 'pollen.net.transport.SingleBufferManager';
    SingleBufferManager.pollen$used = 0;
    SingleBufferManager.Buffer = $units['pollen.net.transport.Buffer'];
    SingleBufferManager.getBuffer = new $$Ref('pollen_net_transport_SingleBufferManager_getBuffer__E');
    SingleBufferManager.getDataLength = new $$Ref('pollen_net_transport_SingleBufferManager_getDataLength__E');
    SingleBufferManager.setDataLength = new $$Ref('pollen_net_transport_SingleBufferManager_setDataLength__E');
    SingleBufferManager.getSize = new $$Ref('pollen_net_transport_SingleBufferManager_getSize__E');
    SingleBufferManager.freeBuffer = new $$Ref('pollen_net_transport_SingleBufferManager_freeBuffer__E');
    SingleBufferManager.getEmptyBuffer = new $$Ref('pollen_net_transport_SingleBufferManager_getEmptyBuffer__E');
    SingleBufferManager.hasEmptyBuffer = new $$Ref('pollen_net_transport_SingleBufferManager_hasEmptyBuffer__E');
    SingleBufferManager.targetInit = new $$Ref('pollen_net_transport_SingleBufferManager_targetInit__I');
    SingleBufferManager.$$hostInit = function() {
        var $$text = '';
        SingleBufferManager.buf2 = new $units['pollen.net.transport.Buffer'].Buffer('pollen_net_transport_SingleBufferManager_buf2__V'); SingleBufferManager.buf2.new_host();
        SingleBufferManager.buf = new $units['pollen.net.transport.Buffer'].Buffer('pollen_net_transport_SingleBufferManager_buf__V');
        (SingleBufferManager.buf) = new $units['pollen.net.transport.Buffer'].Buffer().new_host();
        (SingleBufferManager.bufUsed) = false;
        return $$text;
    }
    SingleBufferManager.pollen__uses__ = function() {
        $$bind($units['pollen.net.transport.Buffer'], 'pollen$used', true);
    }
    SingleBufferManager.$$privateInit = function() {
        SingleBufferManager.bufUsed = 0;
    }
    return SingleBufferManager;
}
var $$u = $$c();
$units['pollen.net.transport.SingleBufferManager'] = $$u;
$units.push($$u);

File: test51_out/pollen.net.transport/SingleBufferManager/SingleBufferManager.c

/*
 * ======== string literals ========
 */

static const char pollen_net_transport_SingleBufferManager__s__fileName[] = "SingleBufferManager.p";
static const char pollen_net_transport_SingleBufferManager__s__0[] = "SingleBufferManager";

/*
 * ======== function definitions ========
 */

uint16 pollen_net_transport_SingleBufferManager_getDataLength__E() {
    return(  pollen_net_transport_Buffer_Buffer_getDataLength__E(&(pollen_net_transport_SingleBufferManager_buf__V))  );
}

uint16 pollen_net_transport_SingleBufferManager_getSize__E() {
    return(  pollen_net_transport_Buffer_Buffer_getSize__E(&(pollen_net_transport_SingleBufferManager_buf__V))  );
}

bool pollen_net_transport_SingleBufferManager_hasEmptyBuffer__E() {
    return(  !(pollen_net_transport_SingleBufferManager_bufUsed__V)  );
}

uint8 *  pollen_net_transport_SingleBufferManager_getBuffer__E() {
    return(  pollen_net_transport_Buffer_Buffer_getBuffer__E(&(pollen_net_transport_SingleBufferManager_buf__V))  );
}

void pollen_net_transport_SingleBufferManager_freeBuffer__E() {
    (pollen_net_transport_SingleBufferManager_bufUsed__V) = false;
}

uint8 *  pollen_net_transport_SingleBufferManager_getEmptyBuffer__E() {
    if (!(pollen_net_transport_SingleBufferManager_bufUsed__V)) {
        (pollen_net_transport_SingleBufferManager_bufUsed__V) = true;
        return(  pollen_net_transport_SingleBufferManager_getBuffer__E()  );
    }
    else {
        return(  null  );
    }
}

void pollen_net_transport_SingleBufferManager_setDataLength__E( uint16 s ) {
    pollen_net_transport_Buffer_Buffer_setDataLength__E(&(pollen_net_transport_SingleBufferManager_buf__V), (s));
}

void pollen_net_transport_SingleBufferManager_targetInit__I() {
}

